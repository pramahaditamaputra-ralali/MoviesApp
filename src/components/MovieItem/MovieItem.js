import React from 'react';
import {Image, StyleSheet, Text, TouchableOpacity, View} from 'react-native';

const MovieItem = ({title, onPress, imgSource}) => {
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <Image
        style={styles.imgBackground}
        source={{
          uri: `https://image.tmdb.org/t/p/w500/${imgSource}`,
        }}
      />

      <View style={styles.movieInfoContainer}>
        <View style={styles.movieTextInfo}>
          <Text>{title}</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    minHeight: 200,
    borderWidth: 1,
    borderColor: '#EAEAEA',
    marginHorizontal: 15,
    marginBottom: 25,
    borderRadius: 5,
  },
  imgBackground: {
    justifyContent: 'flex-end',
    alignItems: 'flex-start',
    resizeMode: 'stretch',
    minHeight: 150,
  },
  movieInfoContainer: {
    minHeight: 50,
    padding: 5,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  movieTextInfo: {},
});

export default MovieItem;
