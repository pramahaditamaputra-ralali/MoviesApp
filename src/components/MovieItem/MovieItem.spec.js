import React from 'react';
import renderer from 'react-test-renderer';
import MovieItem from './MovieItem';

test('renders correctly', () => {
  const tree = renderer.create(<MovieItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
