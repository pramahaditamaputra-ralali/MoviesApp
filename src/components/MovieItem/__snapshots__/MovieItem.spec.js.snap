// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "#EAEAEA",
      "borderRadius": 5,
      "borderWidth": 1,
      "flex": 1,
      "marginBottom": 25,
      "marginHorizontal": 15,
      "minHeight": 200,
      "opacity": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://image.tmdb.org/t/p/w500/undefined",
      }
    }
    style={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "flex-end",
        "minHeight": 150,
        "resizeMode": "stretch",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "minHeight": 50,
        "padding": 5,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text />
    </View>
  </View>
</View>
`;
